packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -60,45 +50,40 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
   protected readonly handlers = {
     // eth_requestAccounts
     handshake: async (_: RequestArguments): Promise<AddressString[]> => {
-      if (this.connected) {
-        this.emit('connect', { chainId: hexStringFromNumber(this.chain.id) });
-        return this.accounts;
+      if (!this.signer) {
+        const signerType = await this.requestSignerSelection();
+        const signer = this.initSigner(signerType);
+        await signer.handshake();
+        this.signer = signer;
+        storeSignerType(signerType);
       }
-
-      const signerType = await this.requestSignerSelection();
-      const signer = this.initSigner(signerType);
-      const accounts = await signer.handshake();
-
-      this.signer = signer;
-      storeSignerType(signerType);
-
-      this.emit('connect', { chainId: hexStringFromNumber(this.chain.id) });
-      return accounts;
+      this.emit('connect', { chainId: hexStringFromNumber(this.signer.chain.id) });
+      return this.signer.accounts;
     },
 
     sign: async (request: RequestArguments) => {
-      if (!this.connected || !this.signer) {
+      if (!this.signer) {
         throw standardErrors.provider.unauthorized(
           "Must call 'eth_requestAccounts' before other methods"
         );
       }
       return await this.signer.request(request);
     },
 
-    fetch: (request: RequestArguments) => fetchRPCRequest(request, this.chain),
+    fetch: (request: RequestArguments) => fetchRPCRequest(request, this.signer?.chain),
```

Comment:
this line reveals that `fetch` handling actually has dependency to signer. we should revisit our design. this is related to what we just discussed @nateReiners